flag.go

```diff
@@ -94,16 +95,82 @@ func bucketUser(user User, key, attr, salt string) float32 {
 	}
 }
 
-func (f FeatureFlag) EvaluateExplain(user User) (interface{}, *Explanation) {
+type EvalResult struct {
+	Value                interface{}
+	Explanation          *Explanation
+	FeatureRequestEvents []FeatureRequestEvent //to be sent to LD
```

Comment:
I think `PrerequisiteRequestEvents` or something more descriptive would be helpful for the attribute name.